src/hooks/useKeyboardNavigation.ts

```diff
@@ -0,0 +1,313 @@
+  // const activeSkinTone = useActiveSkinTone();
+  // const isSkinToneSpreadOpen = useSkinToneSpreadValue();
+
+  useEffect(() => {
+    return tryTinyKeys(categoriesNavRef.current, {
+      ArrowLeft: focusPrevCategory,
+      ArrowRight: focusNextCategory,
+      ArrowDown: focusNextSection,
+      Escape: exitEmojiList
+    });
+  }, []);
+
+  useEffect(() => {
+    if (!emojiSearchRef.current) return;
+
+    return tryTinyKeys(emojiSearchRef.current, {
+      ArrowRight: focusSkinTonePicker,
+      ArrowUp: focusPrevSection,
+      ArrowDown: focusNextSection
+    });
+  }, []);
+
+  useEffect(() => {
+    if (!emojiSearchRef.current) return;
+
+    // return tryTinyKeys(skinToneSpreadRef.current, {
+    //   // ArrowLeft: isSkinToneSpreadOpen ? focusNextSkinTone : exitSkinTones,
+    //   ArrowRight: focusPrevSkinTone,
+    //   Escape: exitSkinTones,
+    //   Enter: exitSkinTones
+    // });
+  }, []);
+
+  useEffect(() => {
+    return tryTinyKeys(emojiListRef.current, {
+      ArrowRight: withPreventDefault(() => navigateGrid(RIGHT)),
+      ArrowLeft: withPreventDefault(() => navigateGrid(LEFT)),
+      ArrowUp: withPreventDefault(() => navigateGrid(UP)),
+      ArrowDown: withPreventDefault(() => navigateGrid(DOWN)),
+      Escape: exitEmojiList
+    });
+  }, []);
+
+  function exitEmojiList() {
+    focusSearch();
+    scrollEmojiListToTop();
+  }
+
+  const updateActiveItem = (newActiveItem: any) => {
+    focusElement(newActiveItem);
+  };
+
+  const navigateGrid = withCatch((direction: any) => {
+    const {
+      activeIndex,
+      itemsPerRow,
+      numOfItems,
+      currentColumn,
+      gridChildren
+    } = getGridInfo();
+
+    const {
+      isTopRow,
+      isBottomRow,
+      isLastRow,
+      isLeftColumn,
+      isRightColumn
+    } = getElementBoundariesInfo({ activeIndex, itemsPerRow, numOfItems });
+
+    switch (direction) {
+      case UP:
+        if (
+          isTopRow &&
+          !focusPrevEmojiListGroup(currentColumn, itemsPerRow ?? -1)
+        )
+          focusPrevSection();
+        else updateActiveItem(gridChildren?.[activeIndex - itemsPerRow]);
+        break;
+      case DOWN:
+        if (isBottomRow) {
+          if (isLastRow) focusNextEmojiListGroup(currentColumn);
+          else updateActiveItem(gridChildren?.[numOfItems - 1]);
+        } else updateActiveItem(gridChildren?.[activeIndex + itemsPerRow]);
+        break;
+      case LEFT: {
+        const prevEmoji = getPrevEmoji();
+
+        if (isLeftColumn)
+          prevEmoji
+            ? focusElement(prevEmoji)
+            : focusPrevEmojiListGroupOnLastItem();
+        else focusElement(prevEmoji);
+
+        break;
+      }
+      case RIGHT: {
+        const nextEmoji = getNextEmoji();
+
+        if (isRightColumn)
+          nextEmoji ? focusElement(nextEmoji) : focusNextEmojiListGroup();
+        else focusElement(nextEmoji);
+
+        break;
+      }
+    }
+  });
+
+  let sections: any[] = [];
```

Comment:
Rather than use any, I think these should satisfy the actual types we use.
```suggestion
type Section = {
  focus: () => void,
  rootElement: React.RefObject<HTMLElement>
};
  let sections: Section[] = [];
```